### PR TITLE
Set OMP_NUM_THREADS=1 in the test suite.

### LINKED ISSUE #4425: PETSc: Make solvers respect deal.II's global thread limit
I knew it was bad - but I never really looked at it how bad it is.
So this is a testsuite via `cmake -j10` on an otherwise empty machine:

load

We have to fix that. Problem is that if a test machine runs a concurrent job this excessive load leads to a number of tests failing due to the 10 minute timeout.

### PR DESCRIPTION
**Pull request by drwells**
This should fix #4425 by exporting `OMP_NUM_THREADS=1` in the shell script that runs the tests.

@kronbichler I reread the discussion in #4425 and you mentioned that we may want to do `OMP_NUM_THREADS=2` instead. Should we do this?

### CODE REVIEW

**Comment by drwells** [comment]
Closes #4425.

**Comment by tamiko** [comment]
Please set the number of threads at least to 2 - that's what I use as a thread limit on `tester` at the moment.

File: tests/base/openmp_threads.cc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#include <cstdlib>
+
+int main()
+{
+  initlog();
+
+  // This might not work on Windows; always pass if we don't have getenv
+  // available
+#ifdef _POSIX_C_SOURCE
+  const char *omp_env_variable = getenv("OMP_NUM_THREADS");
```
Comment (tamiko) [inline tests/base/openmp_threads.cc]: Can you please add an additional guard for openmp itself? (openmp support itself might not necessarily be available…)

**Comment by tamiko** [comment]
@drwells I would prefer to set the corresponding environment variable (if not already set) in the library itself. The rationale behind this approach is the simple fact that you expect a thread-limit set by a deal.II function to be actually binding. That's what I suggested in #4425.

Is there a strong reason to not do that?

File: tests/base/openmp_threads.cc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// Ensure that, when executing test suite programs, the environment variable
+// OMP_NUM_THREADS is set to 1.
```
Comment (bangerth) [inline tests/base/openmp_threads.cc]: you'll have to update that number here as well

**Comment by bangerth** [comment]
I don't think you can do that. `libdeal_II.so` depends on `libpetsc.so`, for example, so the latter is going to be initialized before the former, and definitely before `main()` is run. If PETSc reads the environment variable in the .so startup code, you can't get at that any more afterwards.

File: tests/base/openmp_threads.cc (hunk 1 above)
Comment (tamiko) [inline tests/base/openmp_threads.cc]: @bangerth "OBDA" - "offensichtlich bedingt durch Alkohol"

**Comment by tamiko** [comment]
@bangerth See "Update". I am fine with this patch - I would like to see some mild concurrency preserved, i.e. a thread limit of 2, or similar.

@drwells Thanks for fixing :-)

**Comment by bangerth** [comment]
Yes, a modest (and bounded) concurrency of 2 seems like a good idea to shake out bugs.

File: tests/base/openmp_threads.cc (hunk 1 above)
Comment (drwells) [inline tests/base/openmp_threads.cc]: Aren't you in a timezone an hour behind me anyway?

Comment (bangerth) [inline tests/base/openmp_threads.cc]: He is -- must have been one heck of a night!

(I would have good reason for this kind of behavior though: I was at a Metallica concert in Denver last night till midnight, and the amount of -- legal -- pot in the air was probably sufficient to make me fail any drug test for the next 2 weeks...)

Comment (bangerth) [inline tests/base/openmp_threads.cc]: (I did not actively partake, just for the record.)

**Comment by drwells** [comment]
I updated the patch to export `OMP_NUM_THREADS=2`.

**Pull request merged**